python/socketioclient/socketio_client.py

```diff
@@ -0,0 +1,142 @@
+    #     self.previous_response_times[client_request_id] = time.time()        
+    #     logger.info('Finished call.')
+
+    def on_disconnect(self, *args):
+        logger.info('Server disconnected.' + str(args))
+
+    def on_connect_error(self, *args):
+        logger.info("Cannot connect to the server. Error", args)
+
+    def on_response(self, *args):
+        response_time = time.time()
+        client_request_id = args[0]['client_request_id']
+        logger.info("Client received response: " + str(args))
+        previous_response_time = self.previous_response_times[client_request_id]
+        logger.info(f'Client_request_id: {client_request_id}, time taken for first response {response_time - previous_response_time}')
+        self.previous_response_times[client_request_id] = response_time
+
+    def on_stream(self, *args):
+        response_time = time.time()
+        client_request_id = args[0][0]['client_request_id']
+        previous_response_time = self.previous_response_times[client_request_id]
+        streaming_response_time = response_time - previous_response_time
+        running_avg, num_of_responses = self.running_avg_times.get(client_request_id, (0, 0))
+        running_avg = (running_avg * num_of_responses + streaming_response_time) / (num_of_responses + 1)
+        logger.info("printing arguments ", *args)
+        if num_of_responses % self.print_every_n_response == 0:
+            logger.info(f'client_request_id: {client_request_id}, streaming heart beat: {streaming_response_time}, running average: {running_avg}')
+
+        self.running_avg_times[client_request_id] = (running_avg, num_of_responses + 1)
+        self.previous_response_times[client_request_id] = response_time
+
+    def on_error(self, *args):
+        logger.info("error", args)
+
+
+def main(args):
+    client = FastRFSClient(namespace='/streaming', print_every_n_response=args.print_every_n_response)
+    client.populate_subscription_requests(args.token_pairs, args.levels)
+    headers = create_header(API_KEY, SECRET_KEY, PASSPHRASE)
+    socketio_client = socketio.Client(logger=False, engineio_logger=False, ssl_verify=False)
+    socketio_client.register_namespace(client)
+    socketio_client.connect(URL, namespaces=['/streaming'], transports=['websocket'], headers=headers)
+
+    while True:
+        continue
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--token_pairs',
+                        type=lambda s: s.split(","),
+                        required=False,
+                        default="BTC/USD",
+                        help="Comma separated token pairs eg. BTC/USD,ETH/USD")
+    parser.add_argument('--levels',
+                        type=lambda s: [int(x) for x in s.split(",")],
+                        required=False,
+                        default="1,2",
+                        help="Comma separated levels eg. 190,200,400")
+
+    parser.add_argument('--print_every_n_response',
```

Comment:
lets remove this? Not needed from clients perspective

Reply:
Done!